tcp_handler.c

```diff
@@ -21,22 +21,25 @@ void tcp_network_manager(struct tcp_network_data* nm)
     if (nm->sock < 0)
     {
         ESP_LOGE(TAG, "Unable to create socket: errno %d", errno);
+        tcp_close_network_manager(nm);
```

Comment:
Is it right to close network manager, when it isn't running ?

Reply:
In TCP you need to close it entirely to create a new one to reconnect. So whenever it fails, it closes it and recreates socket.

Reply:
Cool, sounds good